src/hints/state.rs

```diff
@@ -323,6 +326,48 @@ pub fn enter_scope_commitment_info_by_address(
     Ok(())
 }
 
+pub const WRITE_SPLIT_RESULT: &str = indoc! {r#"
+    from starkware.starknet.core.os.data_availability.bls_utils import split
+
+    segments.write_arg(ids.res.address_, split(ids.value))"#
+};
+pub fn write_split_result(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    // this hint fills in a Cairo BigInt3 by taking a felt (ids.value) and passing it to a split fn
+    let value = get_integer_from_var_name(vars::ids::VALUE, vm, ids_data, ap_tracking)?;
+    let res_ptr = get_ptr_from_var_name(vars::ids::RES, vm, ids_data, ap_tracking)?;
+
+    fn split(num: Felt252) -> Vec<Felt252> {
+        // Takes an integer and returns its canonical representation as:
+        //    d0 + d1 * BASE + d2 * BASE**2.
+        // d2 can be in the range (-2**127, 2**127).
+
+        let base: BigInt = BigInt::from(2).pow(86);
+
+        let mut a = Vec::with_capacity(3);
+        let mut num = num.to_bigint();
+        for _ in 1..3 {
```

Comment:
Ranges starting at 1? This is not Matlab :thinking:

Reply:
I swear I didn't write that.

70cc017